libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -346,24 +346,21 @@ bigint StaticAnalyzer::structureSizeEstimate(Type const& _type, set<StructDefini
 	case Type::Category::Array:
 	{
 		auto const& t = dynamic_cast<ArrayType const&>(_type);
-		return structureSizeEstimate(*t.baseType(), _structsSeen) * (t.isDynamicallySized() ? 1 : t.length());
+		if (!t.isDynamicallySized())
+			return structureSizeEstimate(*t.baseType(), _structsSeen) * t.length();
+		break;
```

Comment:
Note to self: this change may be summarized as follows

- Estimate size of dynamically sized arrays as `1` instead of `structureSizeEstimate(<arrayBaseType> * 1)` i.e., this change ignores the base array type for dynamically sized arrays
- Retain size estimate of statically sized arrays